src/rules/import-alias.ts

```diff
@@ -10,23 +10,72 @@ import type { JSONSchema4 } from "json-schema";
 import { dirname, join as joinPath, resolve, sep as pathSep } from "path";
 import slash from "slash";
 
+interface ConfigMatch {
+    type: "pattern" | "path";
+    length: number;
+    depth: number;
+}
+
 function isPermittedRelativeImport(
     importModuleName: string,
     relativeImportOverrides: RelativeImportConfig[],
-    fileAbsoluteDir: string
+    fileAbsoluteDir: string,
+    filepath: string
 ) {
-    const configOfPath = relativeImportOverrides.find((config) =>
-        fileAbsoluteDir.includes(resolve(config.path))
-    );
-    if (!configOfPath) {
-        return false;
-    }
-
     const importParts = importModuleName.split("/");
     const relativeDepth = importParts.filter(
         (moduleNamePart) => moduleNamePart === ".."
     ).length;
-    return configOfPath.depth >= relativeDepth;
+
+    const pathsMatchs: ConfigMatch[] = [];
+    const patternsMatchs: ConfigMatch[] = [];
+
+    relativeImportOverrides.filter((config) => {
+        if (config.pattern) {
+            const regex = new RegExp(config.pattern);
+            console.log("regex", regex, filepath);
+            console.log("regex2", regex.test(filepath));
+            if (regex.test && regex.test(filepath)) {
+                patternsMatchs.push({
+                    type: "pattern",
+                    length: config.pattern.length,
+                    depth: config.depth,
+                });
+            }
+        } else {
+            if (fileAbsoluteDir.includes(resolve(config.path))) {
+                pathsMatchs.push({
+                    type: "path",
+                    length: config.path.length,
+                    depth: config.depth,
+                });
+            }
+        }
+    });
+
+    if (!pathsMatchs.length && !patternsMatchs.length) {
+        return false;
+    }
+
+    // sorting mainly ensure that errors are checked from highest rule to lowest
+
+    const sortedPathsmatchs = pathsMatchs.sort((a, b) => a.length - b.length);
```

Comment:
Sorting is probably a good idea, but let's keep it simple and do it by `depth`, with highest depth as the top priority. We'll rely on users setting their configurations correctly instead of defining a behavior for them like paths being preferred over patterns.